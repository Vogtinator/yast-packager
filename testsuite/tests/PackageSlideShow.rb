# encoding: utf-8

# Testsuite for PackageSlideShow.ycp module
#
# $Id:$
module Yast
  class PackageSlideShowClient < Client
    def main
      Yast.include self, "testsuite.rb"
      @READ = { "target" => { "tmpdir" => "/tmp" } }
      TESTSUITE_INIT([@READ], nil)

      Yast.import "PackageSlideShow"

      # no cut off
      TEST(lambda { PackageSlideShow.ListSumCutOff([60, 70, 80, 0], 100) }, [], nil)

      # one cut off
      TEST(lambda { PackageSlideShow.ListSumCutOff([60, 70, 80, 150], 100) }, [], nil)

      # more cut offs
      TEST(lambda { PackageSlideShow.ListSumCutOff([160, 170, 180, 10], 100) }, [], nil)

      nil
    end
  end
end

Yast::PackageSlideShowClient.new.main
