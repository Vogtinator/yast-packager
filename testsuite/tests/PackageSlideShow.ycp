/**
 * Testsuite for PackageSlideShow.ycp module
 *
 * $Id:$
 */

{

include "testsuite.ycp";
map READ = $[
  "target" : $[
    "tmpdir" : "/tmp"
  ]
];
TESTSUITE_INIT ([READ], nil);

import "PackageSlideShow";

// no cut off
TEST(``(PackageSlideShow::ListSumCutOff([60, 70, 80, 0], 100)), [], nil);

// one cut off
TEST(``(PackageSlideShow::ListSumCutOff([60, 70, 80, 150], 100)), [], nil);

// more cut offs
TEST(``(PackageSlideShow::ListSumCutOff([160, 170, 180, 10], 100)), [], nil);
}
