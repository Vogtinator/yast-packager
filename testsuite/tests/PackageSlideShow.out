Read	.target.tmpdir "/tmp"
Return	210
Return	-100
Return	-300
